Source/CarthageKit/Xcode.swift

```diff
@@ -992,6 +995,15 @@ public func buildInDirectory(directoryURL: NSURL, withConfiguration configuratio
 							|> map { _ in scheme }
 					}
 					|> collect
+					|> catch { error in
+						switch error {
+						case .NoSharedSchemes:
+							return .empty
+
+						default:
+							return SignalProducer(error: error)
+						}
+					}
```

Comment:
In other words, #570 works well if some projects does not have a shared framework scheme, but have some other shared schemes. If those projects have no any shared schemes, `buildInDirectory()` errors out at the first project locator of them.